src/app/events/[slug]/components/SponsorSection.tsx

```diff
@@ -0,0 +1,73 @@
+'use client'
+
+import type { Event } from '@/types/eventType'
+
+import { buildImageSizeProp } from '@/utils/buildImageSizeProp'
+
+import { PageSection } from '@/components/PageSection'
+
+import { SponsorGrid } from './SponsorGrid'
+
+export type Sponsor = NonNullable<
+  NonNullable<Event['sponsors']>['first-tier']
+>[number]
+
+type SponsorSectionProps = {
+  sponsors: Event['sponsors']
+}
+
+const sponsorTierConfigs = {
+  'first-tier': {
+    gridClassName: 'grid-cols-2 md:grid-cols-4',
+    logoImageConfig: {
+      containerStyles: 'h-16 sm:h-20',
+      imageSizes: buildImageSizeProp({ startSize: '140px', sm: '180px' }),
+    },
+  },
+  'second-tier': {
+    gridClassName: 'grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5',
+    logoImageConfig: {
+      containerStyles: 'h-11 sm:h-16',
+      imageSizes: buildImageSizeProp({ startSize: '100px', sm: '140px' }),
+    },
+  },
+  'third-tier': {
+    gridClassName: 'grid-cols-4 sm:grid-cols-5 md:grid-cols-5 lg:grid-cols-7',
+    logoImageConfig: {
+      containerStyles: 'h-7 sm:h-11',
+      imageSizes: buildImageSizeProp({ startSize: '60px', sm: '100px' }),
+    },
+  },
+}
+
+const tiers: Array<keyof typeof sponsorTierConfigs> = [
+  'first-tier',
+  'second-tier',
+  'third-tier',
+]
+
+export function SponsorSection({ sponsors }: SponsorSectionProps) {
+  if (!sponsors) return null
+
+  const visibleTiers = tiers.filter((tier) => sponsors[tier])
```

Comment:
I find the word "visible" might imply that there's some visual state - maybe `populatedTiers` or `activeTiers`? What do you think?

Reply:
Actually not sure what you think of this approach? If we include the key tier in the `sponsorTierConfig`, we could eliminate the `tiers` const and `visibleTiers` filtering

```const sponsorTierConfigs = [
  {
    tier: 'first-tier',
    gridClassName: 'grid-cols-2 md:grid-cols-4',
    logoImageConfig: {
      containerStyles: 'h-16 sm:h-20',
      imageSizes: buildImageSizeProp({ startSize: '140px', sm: '180px' }),
    },
  },
  {
    tier: 'second-tier',
    gridClassName: 'grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5',
    logoImageConfig: {
      containerStyles: 'h-11 sm:h-16',
      imageSizes: buildImageSizeProp({ startSize: '100px', sm: '140px' }),
    },
  },
  {
    tier: 'third-tier',
    gridClassName: 'grid-cols-4 sm:grid-cols-5 md:grid-cols-5 lg:grid-cols-7',
    logoImageConfig: {
      containerStyles: 'h-7 sm:h-11',
      imageSizes: buildImageSizeProp({ startSize: '60px', sm: '100px' }),
    },
  },
] as const

export function SponsorSection({ sponsors }: SponsorSectionProps) {
  if (!sponsors) return null

  return (
    <PageSection kicker="sponsors" title="Sponsors">
      <div className="grid gap-8">
        {sponsorTierConfigs.map(({ tier, gridClassName, logoImageConfig }) => {
          if (!sponsors[tier]) return null

          return (
            <SponsorGrid
              key={tier}
              sponsors={sponsors[tier]!}
              tier={tier}
              gridClassName={gridClassName}
              logoImageConfig={logoImageConfig}
            />
          )
        })}
      </div>
    </PageSection>
  )
}
```

Reply:
Love it! Done.